clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -1518,7 +1526,7 @@ NamedDecl *Sema::ActOnNonTypeTemplateParameter(Scope *S, Declarator &D,
 
   QualType T = CheckNonTypeTemplateParameterType(TInfo, D.getIdentifierLoc());
   if (T.isNull()) {
-    T = Context.IntTy; // Recover with an 'int' type.
+    T = TInfo->getType();
```

Comment:
In what sort of context does this matter? Does the effect of this show up in any test?

Is it possible we might recover with something strange which will have weird effects or crash later, like undeduced auto?

Reply:
We PROBABLY should do some work here to make sure that `TInfo->getType` is a valid type/is also not null.  So something like:
`T = TInfo->getType()->isNull() ? Context.IntTy : TInfo->getType();`